### PR TITLE
Doc update HashSet

### PR DESCRIPTION
**Pull request by jpsiitonen**

### CODE REVIEW

File: sdk/lib/collection/hash_set.dart [hunk 1]
```diff
@@ -21,6 +21,87 @@ part of dart.collection;
...
+/// final String elementAt = hashSet.elementAt(1);
+/// print(elementAt); // D
+///  ```
+/// The [forEach] iterates through all entries of a set.
+/// Manipulating item count in [forEach] is prohibited. Adding or
+/// deleting items during iteration causes an exception:
+/// _"Concurrent modification during iteration"_.
+/// ```dart
+/// hashSet.forEach((element) {
+///   print(element);
```
Comment (alexlindroos) [inline sdk/lib/collection/hash_set.dart:53]: missing the result

File: sdk/lib/collection/hash_set.dart [hunk 2]
```diff
@@ -21,6 +21,87 @@ part of dart.collection;
...
+/// ```
+/// To add item to set, call [add]
+/// ```dart
+/// hashSet.add('E');
+/// print(hashSet); // // {A, D, C, E, B}
+/// ```
+/// To remove specific value, call [remove]:
+/// ```dart
+/// hashSet.remove('A');
+/// print(hashSet); // // {D, C, E, B}
```
Comment (alexlindroos) [inline sdk/lib/collection/hash_set.dart]: double //

File: sdk/lib/collection/hash_set.dart [hunk 3]
```diff
@@ -21,6 +21,87 @@ part of dart.collection;
...
 /// multiple iterations over the same set produce the same order, as long as
 /// the set is not modified.
+///
+/// Example:
+/// ```dart
+/// final HashSet hashSet = HashSet();
+/// hashSet.addAll({'A', 'B', 'C', 'D'});
+/// hashSet.isEmpty; // false
+/// hashSet.length; // 4
+/// print(hashSet); / {A, D, C, B}
```
Comment (alexlindroos) [inline sdk/lib/collection/hash_set.dart]: Only one /

Comment (alexlindroos) [reply sdk/lib/collection/hash_set.dart]: Is the result random? For me it was A,B,C,D

Comment (jpsiitonen) [reply sdk/lib/collection/hash_set.dart]: the hashset iteration can happen in any order.

File: sdk/lib/collection/hash_set.dart [hunk 4]
```diff
@@ -17,10 +17,81 @@ part of dart.collection;
...
+/// final copyOfOriginal = hashSet.toSet();
+/// print(copyOfOriginal); // {A, C, D, B}
+///
+/// // To add item to set, call [add]
+/// hashSet.add('E');
+/// print(hashSet); // {A, D, C, E, B}
+///
+/// // To remove specific value, call remove
+/// hashSet.remove('A');
+/// print(hashSet); // // {D, C, E, B}
```
Comment (alexlindroos) [inline sdk/lib/collection/hash_set.dart]: double //

**Review by alexlindroos — APPROVED** [review]

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**